### PR TITLE
Add base HttpResponse

### PR DESCRIPTION
**Pull request by RikkiGibson**
Related to Azure/autorest.typescript#237

### CODE REVIEW

File: lib/httpOperationResponse.ts [hunk 1]
```diff
@@ -5,10 +5,9 @@ import { WebResource } from "./webResource";
 import { HttpHeaders } from "./httpHeaders";
 
 /**
- * Wrapper object for http request and response. Deserialized object is stored in
- * the `parsedBody` property when the response body is received in JSON or XML.
+ * The properties on an HTTP response which will always be present.
  */
-export interface HttpOperationResponse<TBody = any, THeaders = any> {
+export interface BaseHttpOperationResponse {
```
Comment (daschult) [inline lib/httpOperationResponse.ts]: We previously thought about renaming HttpOperationResponse to HttpResponse. Would this be a good time to do that, rather that adding the "Base" prefix?

Comment (RikkiGibson) [reply lib/httpOperationResponse.ts]: So you think HttpResponse as the base, HttpOperationResponse as the "bucket of all the optionals"?

Comment (daschult) [reply lib/httpOperationResponse.ts]: Yeah. Your design seems to say that HttpOperationResponse is being kept around just for backwards compatability and that BaseHttpOperationResponse/HttpResponse is what should be used/extended moving forward.

**Comment by RikkiGibson** [comment]
BTW, I'm removing the type parameters in HttpOperationResponse, so I'm bumping the minor version

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "email": "azsdkteam@microsoft.com",
     "url": "https://github.com/Azure/ms-rest-js"
   },
-  "version": "0.18.0",
+  "version": "0.19.0",
```
Comment (RikkiGibson) [inline package.json]: Ahh, forgot to update the changelog again. I might just grind through the old versions to fill out the changelog so we can start making a habit of it.

**Pull request merged**